src/Logger.js

```diff
@@ -0,0 +1,125 @@
+        this._logger       = this._createLogger(loglevel);
+        this._accessLogger = null;
+        this.setContext(context);
+    }
+
+    /**
+     * Sets the pino-compatible log level to be used from this point forward.
+     *
+     * @param {string} loglevel  Accepted values: fatal | error | warn | info | debug | trace
+     */
+    setLogLevel(loglevel) {
+        this._logger.level = loglevel;
+    }
+
+    /**
+     * Sets common props that will be added to all logs from this point forward.
+     * Useful for things like request_id, type/namespace/service, etc.
+     *
+     * @param {object} context   Common props to add to all logged messages.
+     */
+    setContext(context) {
+        this._context = context;
+    }
+
+    /**
+     * Adds/overwrites common props that will be added to all logs from this point forward.
+     * Useful for things like request_id, type/namespace/service, etc.
+     *
+     * @param {object} context   Common props to add to all logged messages.
+     */
+    addContext(context) {
+        this._context = {...this._context, ...context};
+    }
+
+    /**
+     * Creates a child logger from the current one, inherting all of parent logger context.
+     *
+     * @param {object} context   Common props to add to current context.
+     */
+    child(context) {
+        return new Logger(this._logger.level, {...this._context, ...context});
+    }
+
+    info(message, props = {}) {
+        this._log("info", message, props);
+    }
+
+    warn(message, props = {}) {
+        this._log("warn", message, props);
+    }
+
+    error(message, props = {}) {
+        if (props instanceof Error) {
+            var error_props = {
+                error_message: props.message
+            };
+            if (props.hasOwnProperty("code")) {
+                error_props.error_code = props.code;
+            }
+            if (props.hasOwnProperty("status")) {
+                error_props.error_status = props.status;
+            }
+            if (props.hasOwnProperty("stack")) {
+                error_props.error_stack = props.stack;
+            }
+            props = error_props;
+        }
+        this._log("error", message, props);
+    }
+
+    debug(message, props = {}) {
+        this._log("debug", message, props);
+    }
+
+    /**
+     * Helper method to record access logs. We want these with no message, context: "access" and
+     * not casting all props as strings, since we use a few integers there and the format is standard.
+     */
+    access(props = {}) {
+        // We have a separate logger for access logs (only if our app needs to) since we
+        // log them as INFO but we want them written regardless of logger log level set.
+        if (null === this._accessLogger) {
+            this._accessLogger = this._createLogger("info");
+        }
+        props = {...this._context, ...props, loglevel: "INFO", context: "access"};
+
+        this._accessLogger.info({...this._context, ...props});
```

Comment:
A couple of feedbacks on this: 

1) I'm not sure this is correct. It seems you're applying `props` twice

```
props = {...this._context, ...props, loglevel: "INFO", context: "access"};
this._accessLogger.info({...this._context, ...props});
```

Should be just:

```
this._accessLogger.info({...this._context, ...props, loglevel: "INFO", context: "access"});
```

2) Should we really apply `this._context` to access logs? The context is used to add info to application logs and I believe if we use here we're just going to add unnecessary fields to access logs, that have a very specific format and shouldn't be polluted by other informations.

Reply:
Well at the end of the day it depends on how you use it.. for the adstitcher for example, we really don't set this context for all logged messages including access logs: `request_id`, `http_method` and `http_uri`. So those should be present as well. Yes, we can also easily adapt the code so that we also pass those props manually when we call `.access()` instead of relying on this context. But I think it's a bit counterintuitive that context gets applied to all logs except access logs. I would keep this behavior.